### ISSUE TITLE
[Bug] az login (used for az devops) not able to retrieve token for user assigned identities

### ISSUE THREAD

**Issue by kartheek0274** [Azure/azure-devops-cli-extension#929]
**Describe the bug**
I am trying to use az login --identity -u (user assigned identity) for logging in azure client. And later I am using az artifacts download to download a universal package. The artifacts download fails with not able to retrieve token from az login.

From azure devops wiki, it looks it supports logging in through az login. I could easily login using my az login credentials, but when i try with identity az devops is not able to get a token. I am trying to create an automated pipeline to download packages. PAT seems like the other route where we can work direclty with az devops login, but i dont want the access token to tied to a particular individual. It would be great if we can get unblocked using an identity. Let me know if there is something i am missing in my steps.

**To Reproduce**
Azure Cli Version: 2.0.78
Use az --version 
Look for something like-
azure-cli (2.0.70)

Azure-Devops extension version:  0.16.0
Use az --version
Look for something like-
Extensions:
azure-devops (0.16.0)

1. login using az login
2. setup defaults..
3. Run command....

**Expected behavior**

**Screenshots**

**Debug logs**
Add the output of the command running it with debug mode (--debug)
PS C:\Users\{user}> az login --identity -u /subscriptions/{subscription}/resourcegroups/{resourcegroup}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identity}
[
  {
    "environmentName": "AzureCloud",
    "id": "{subscription}",
    "isDefault": true,
    "name": "{subscriptionname}",
    "state": "Enabled",
    "tenantId": "{tenantId}",
    "user": {
      "assignedIdentityInfo": "MSIResource-/subscriptions/{subscriptionId}/resourcegroups/{resourcegroup}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identity}",
      "name": "userAssignedIdentity",
      "type": "servicePrincipal"
    }
  }
]

PS C:\Users\arcadiauser> az artifacts universal download --organization https://msdata.visualstudio.com/ --project="{project}" --scope project --feed {feedname} --name my-first-package --version 0.0.1 --path . --debug

EBUG: trying to get token (temp) for tenant {mytenant} and user userAssignedIdentity
EBUG: adal-python : 83652d7c-fe97-4633-9a55-7f65cc3a6961 - Authority:Performing instance discovery: ...
EBUG: adal-python : 83652d7c-fe97-4633-9a55-7f65cc3a6961 - Authority:Performing static instance discovery
EBUG: adal-python : 83652d7c-fe97-4633-9a55-7f65cc3a6961 - Authority:Authority validated via static instance discovery
EBUG: adal-python : 83652d7c-fe97-4633-9a55-7f65cc3a6961 - TokenRequest:Getting token from cache with refresh if necessary.
EBUG: adal-python : 83652d7c-fe97-4633-9a55-7f65cc3a6961 - CacheDriver:finding with query keys: {'_clientId': '...', 'userId': '...'}
EBUG: adal-python : 83652d7c-fe97-4633-9a55-7f65cc3a6961 - CacheDriver:Looking for potential cache entries: {'_clientId': '...', 'userId': '...'}
EBUG: adal-python : 83652d7c-fe97-4633-9a55-7f65cc3a6961 - CacheDriver:Found 0 potential entries.
**EBUG: not able to get token from az login
EBUG: Could not retrieve token from local cache. Please run 'az login'.**
raceback (most recent call last):
 File "C:\Users\arcadiauser\.azure\cliextensions\azure-devops\azext_devops\dev\common\services.py", line 157, in get_token_from_az_login
   auth_token = profile.get_access_token_for_resource(user, tenant, '499b84ac-1321-427f-aa17-267ca6975798')
 File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-2ombfg7z\azure-cli-core\azure\cli\core\_profile.py", line 509, in get_access_token_for_resource
 File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-2ombfg7z\azure-cli-core\azure\cli\core\_profile.py", line 919, in retrieve_token_for_user
nack.util.CLIError: Could not retrieve token from local cache. Please run 'az login'.
EBUG: Getting credential: azdevops-cli:https://msdata.visualstudio.com
EBUG: Keyring backend : keyring.backends.Windows.WinVaultKeyring (priority: 5)
EBUG: Getting credential: azdevops-cli: default
EBUG: Keyring backend : keyring.backends.Windows.WinVaultKeyring (priority: 5)
EBUG: Before you can run Azure DevOps commands, you need to run the login command(az login if using AAD/MSA identity else az devops login if using PAT token) to setup credentials.  Please see
ttps://aka.ms/azure-devops-cli-auth for more information.
raceback (most recent call last):
 File "C:\Users\arcadiauser\.azure\cliextensions\azure-devops\azext_devops\dev\common\artifacttool_updater.py", line 58, in _get_artifacttool
   release = _get_current_release(organization, override_version)
 File "C:\Users\arcadiauser\.azure\cliextensions\azure-devops\azext_devops\dev\common\artifacttool_updater.py", line 137, in _get_current_release
   connection = get_connection(organization)
 File "C:\Users\arcadiauser\.azure\cliextensions\azure-devops\azext_devops\dev\common\services.py", line 43, in get_connection
   credentials = _get_credentials(organization)
 File "C:\Users\arcadiauser\.azure\cliextensions\azure-devops\azext_devops\dev\common\services.py", line 81, in _get_credentials
   raise get_authentication_error('Before you can run Azure DevOps commands, you need to run the login command'
nack.util.CLIError: Before you can run Azure DevOps commands, you need to run the login command(az login if using AAD/MSA identity else az devops login if using PAT token) to setup credentials.  Please
ee https://aka.ms/azure-devops-cli-auth for more information.
EBUG: handling generic error

**Additional context**

**Comment by atbagga** [comment]
@kartheek0274  Unfortunately Azure-Devops does not support login using service principals and hence nor does the azure-devops CLI extension. 

For now PAT is your only option for automation. 

Here is a open issue for login through service-principals - https://github.com/Azure/azure-devops-cli-extension/issues/905. So this issue is effectively a duplicate of the linked feature request.

**Comment by kartheek0274** [comment]
@atbagga , thanks for the information. We are looking to automate this process for compliant projects to make sure that we dont get tied up ourselves with PATs. What is the recommended approach for automation?. How are other teams doing this?.

**Comment by geverghe** [comment]
@kartheek0274 - For now, as @atbagga mentioned, the only supported automation approach is to use PAT. Based on anecdotal evidence, a service account PAT scoped to the required permission is used for automation by most teams.

**Comment by atbagga** [comment]
Closing as duplicate of #905

**Comment by pribylmartinnet** [comment]
I used this Tutorials https://docs.microsoft.com/en-us/azure/devops-project/devops-starter-gh-web-app ans I get error: Run azure/CLI@v1
Starting script execution via docker image mcr.microsoft.com/azure-cli:latest
ERROR: Could not retrieve credential from local cache for service principal d1557a2-9375-47ce-92e2-6c0f7ab34c3f. Run `az login` for this service principal.
Error: Error: az cli script failed.
cleaning up container...
MICROSOFT_AZURE_CLI_163640042399_CONTAINER
